lib/Doctrine/Common/Cache/FileCache.php

```diff
@@ -78,8 +109,60 @@ public function __construct($directory, $extension = '')
             ));
         }
 
+        $extension = (string) $extension;
         $this->directory = realpath($directory);
-        $this->extension = (string) $extension;
+        $this->extension = $extension ?: $this->extension;
+        $this->fileMode = 0666 & ~umask();
+    }
+
+    /**
+     * Sets the mode that new directories will be created with.
+     * @param int $mode Mode, normally in octal (ex: 0755)
+     */
+    public function setDirectoryMode($mode)
```

Comment:
do we need a setter for this ? the directory mode will end up being 0777 minus the umask because of the ``mkdir`` behavior (and if you configure it as 0775, it will still actually be 0755 minus the umask). I think it is easier to assume that you can set the umask to the value you need if you have special needs (this is what Symfony assumes for instance)